sites/public/__tests__/pages/applications/contact/address.test.tsx

```diff
@@ -0,0 +1,72 @@
+import React from "react"
+import { setupServer } from "msw/lib/node"
+import { fireEvent } from "@testing-library/react"
+import { mockNextRouter, render } from "../../../testUtils"
+import ApplicationAddress from "../../../../src/pages/applications/contact/address"
+
+window.scrollTo = jest.fn()
+
+const server = setupServer()
+
+beforeAll(() => {
+  server.listen()
+  mockNextRouter()
+})
+
+afterEach(() => server.resetHandlers())
+
+afterAll(() => server.close())
+
+describe("applications pages", () => {
+  afterAll(() => {
+    jest.clearAllMocks()
+  })
+
+  describe("address step", () => {
+    it("should render form fields", () => {
+      const { getByText, getByTestId, getAllByTestId } = render(<ApplicationAddress />)
+
+      expect(
+        getByText("Now we need to know how to contact you about your application.", {
+          exact: false,
+        })
+      ).toBeInTheDocument()
+      expect(getByTestId("app-primary-phone-number")).toBeInTheDocument()
+      expect(getByTestId("app-primary-phone-number-type")).toBeInTheDocument()
+      expect(getByTestId("app-primary-no-phone")).toBeInTheDocument()
+      expect(getByTestId("app-primary-additional-phone")).toBeInTheDocument()
+      expect(getByTestId("app-primary-address-street")).toBeInTheDocument()
+      expect(getByTestId("app-primary-address-street2")).toBeInTheDocument()
+      expect(getByTestId("app-primary-address-city")).toBeInTheDocument()
+      expect(getByTestId("app-primary-address-state")).toBeInTheDocument()
+      expect(getByTestId("app-primary-address-zip")).toBeInTheDocument()
+      expect(getByTestId("app-primary-send-to-mailing")).toBeInTheDocument()
+      expect(getAllByTestId("app-primary-contact-preference")).toHaveLength(4)
+      expect(getByTestId("app-primary-work-in-region-yes")).toBeInTheDocument()
+      expect(getByTestId("app-primary-work-in-region-no")).toBeInTheDocument()
```

Comment:
getByTestId works correctly. However, to get the most accurate experience a user would have, especially when considering accessibility it should be the last choice. https://testing-library.com/docs/queries/about/#priority. For form fields we should be using `getByLabelText`. For example instead of ` expect(getByTestId("app-primary-phone-number-type")).toBeInTheDocument()` we could do `expect(getByLabelText("What type of number is this?")).toBeInTheDocument()`. That way we not only test that the field exist, but also that we are displaying the correct language and it has a proper label tag.

Reply:
Great point! Would love to fast-follow with that